awx_collection/test/awx/test_module_utils.py

```diff
@@ -1,11 +1,66 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+import json
 import sys
 
+from requests.models import Response
 from unittest import mock
 
-import json
+
+def getheader(self, header_name, default):
+    mock_headers = {'X-API-Product-Name': 'not-junk', 'X-API-Product-Version': '1.2.3'}
+    return mock_headers.get(header_name, default)
+
+
+def read(self):
+    return json.dumps({})
+
+
+def status(self):
+    return 200
+
+
+def mock_ping_response(self, method, url, **kwargs):
+    r = Response()
+    r.getheader = getheader.__get__(r)
+    r.read = read.__get__(r)
+    r.status = status.__get__(r)
+    return r
+
+
+def test_version_warning(collection_import, silence_warning):
+    TowerModule = collection_import('plugins.module_utils.tower_api').TowerModule
+    cli_data = {'ANSIBLE_MODULE_ARGS': {}}
+    testargs = ['module_file2.py', json.dumps(cli_data)]
+    with mock.patch('ansible.module_utils.basic.AnsibleModule.warn') as mock_warn:
+        with mock.patch.object(sys, 'argv', testargs):
+            with mock.patch('ansible.module_utils.urls.Request.open', new=mock_ping_response):
+                my_module = TowerModule(argument_spec=dict())
+                my_module._COLLECTION_VERSION = "1.0.0"
+                my_module._COLLECTION_TYPE = "not-junk"
+                my_module.collection_to_version['not-junk'] = 'not-junk'
+                my_module.get_endpoint('ping')
+    mock_warn.assert_called_once_with(
```

Comment:
here also, the warn method seems to be double-mocked, with `silence_warning` doing the same thing that `mock_warn` does. Not surprised that it works, because the last mock wins, but better to just mock once.